modules/xfem/src/base/XFEM.C

```diff
@@ -1507,3 +1557,162 @@ XFEM::getXFEMqRuleOnSurface(std::vector<Point> & intersection_points, std::vecto
     }
   }
 }
+
+void
+XFEM::saveSolutionForNode(const Node * node_to_store_to,
+                          const Node * node_to_store_from,
+                          SystemBase & sys,
+                          std::map<unique_id_type, std::vector<Real>> & stored_solution,
+                          const NumericVector<Number> & current_solution,
+                          const NumericVector<Number> & old_solution)
+{
+  const std::vector<MooseVariable *> & vars = sys.getVariables(0);
+  _stored_solution_scratch.clear();
+  _stored_solution_scratch.reserve(2 * vars.size());
```

Comment:
This is pretty special logic here and isn't documented. It looks like you are doubling storage so you can store current and old values. However you don't say the order in which you store this information. Also, why not store older values? Will that ever be necessary?